cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -183,26 +185,32 @@ public void testServiceManager() throws Exception {
 
   @Test
   public void testAppConfigWithNull() throws Exception {
-    testAppConfig(null);
+    testAppConfig(deployApplication(ConfigTestApp.class), null);
   }
 
   @Test
   public void testAppConfig() throws Exception {
-    testAppConfig(new ConfigTestApp.ConfigClass("testStream", "testDataset"));
+    ConfigTestApp.ConfigClass conf = new ConfigTestApp.ConfigClass("testStream", "testDataset");
+    testAppConfig(deployApplication(ConfigTestApp.class, conf), conf);
   }
 
-  private void testAppConfig(ConfigTestApp.ConfigClass object) throws Exception {
-    String streamName = ConfigTestApp.DEFAULT_STREAM;
-    String datasetName = ConfigTestApp.DEFAULT_TABLE;
-
-    ApplicationManager appManager;
-    if (object != null) {
-      streamName = object.getStreamName();
-      datasetName = object.getTableName();
-      appManager = deployApplication(ConfigTestApp.class, object);
-    } else {
-      appManager = deployApplication(ConfigTestApp.class);
-    }
+  @Test
+  public void testAppFromArtifact() throws Exception {
```

Comment:
Will the current methods to deployApp in TestFramework still remain? If yes, what will be the version of the artifacts created from it? Is the plan to have both the current and new methods exist together?